pkg/resources/envoy/deployment.go

```diff
@@ -58,32 +69,35 @@ func (r *Reconciler) deployment(log logr.Logger, extListener v1beta1.ExternalLis
 	}
 
 	return &appsv1.Deployment{
-		ObjectMeta: templates.ObjectMeta(
-			fmt.Sprintf(envoyDeploymentName, extListener.Name, r.KafkaCluster.GetName()),
-			labelsForEnvoyIngress(r.KafkaCluster.GetName(), extListener.Name), r.KafkaCluster),
+		ObjectMeta: templates.ObjectMetaWithAnnotations(
+			deploymentName,
+			labelsForEnvoyIngress(r.KafkaCluster.GetName(), annotationName),
+			ingressConfig.EnvoyConfig.GetAnnotations(),
+			r.KafkaCluster),
 		Spec: appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{
-				MatchLabels: labelsForEnvoyIngress(r.KafkaCluster.GetName(), extListener.Name),
+				MatchLabels: labelsForEnvoyIngress(r.KafkaCluster.GetName(), annotationName),
 			},
-			Replicas: util.Int32Pointer(r.KafkaCluster.Spec.EnvoyConfig.GetReplicas()),
+			Replicas: util.Int32Pointer(ingressConfig.EnvoyConfig.GetReplicas()),
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
-					Labels:      labelsForEnvoyIngress(r.KafkaCluster.GetName(), extListener.Name),
-					Annotations: generatePodAnnotations(r.KafkaCluster, extListener, log),
+					Labels: labelsForEnvoyIngress(r.KafkaCluster.GetName(), annotationName),
```

Comment:
Why `annotationName` is passed into `labelsForEnvoyIngress` instead of `extListener.Name`?